### PR TITLE
Fix various phpstan errors

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

File: src/Extension/ExtensionInterface.php [hunk 1]
```diff
@@ -68,8 +67,8 @@ public function getFunctions();
      * @return array<array> First array of unary operators, second array of binary operators
      *
      * @psalm-return array{
-     *     array<string, array{precedence: int, class: class-string<AbstractUnary>}>,
-     *     array<string, array{precedence: int, class: class-string<AbstractBinary>, associativity: ExpressionParser::OPERATOR_*}>
+     *     array<string, array{precedence: int, class: class-string<AbstractExpression>}>,
+     *     array<string, array{precedence: int, class?: class-string<AbstractExpression>, associativity: ExpressionParser::OPERATOR_*}>
```
Comment (stof) [inline src/Extension/ExtensionInterface.php:71]: why is the class optional for binary operators but mandatory for unary ones ?

And should we really widen the types there ?

File: src/Runtime/EscaperRuntime.php [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 final class EscaperRuntime implements RuntimeExtensionInterface
 {
+    /** @var array<callable(string $string, string $charset): string> */
```
Comment (stof) [inline src/Runtime/EscaperRuntime.php]:
```suggestion
    /** @var array<string, callable(string $string, string $charset): string> */
```

This way, phpstan will be aware of the type of array keys

File: src/Runtime/EscaperRuntime.php [hunk 3]
```diff
@@ -46,7 +47,7 @@ public function setEscaper($strategy, callable $callable)
     /**
      * Gets all defined escapers.
      *
-     * @return array<callable(string $string, string $charset)> An array of escapers
+     * @return array<callable(string $string, string $charset): string> An array of escapers
```
Comment (stof) [inline src/Runtime/EscaperRuntime.php]:
```suggestion
     * @return array<string, callable(string $string, string $charset): string> An array of escapers
```

File: src/Test/NodeTestCase.php [hunk 4]
```diff
@@ -53,7 +48,7 @@ protected function getCompiler(?Environment $environment = null)
 
     protected function getEnvironment()
     {
-        return $this->currentEnv = new Environment(new ArrayLoader());
```
Comment (stof) [inline src/Test/NodeTestCase.php:56]: maybe it was meant to be `??=` instead, to reuse the same instance when called multiple times in a test

File: src/Extension/ExtensionInterface.php (hunk 1 above)
Comment (fabpot) [reply src/Extension/ExtensionInterface.php:71]: Yes, we do want to widen the types. AbstractUnary and AbstractBinary have nothing special, the only thing we care about is the fact they are AbstractExpression.

For `??`, the class is not an AbstractBinary and that's fine:

```
'??' => ['precedence' => 300, 'class' => NullCoalesceExpression::class, 'associativity' => ExpressionParser::OPERATOR_RIGHT],
```

For `is` and `is not`, we don't have a class as it's handled by `ExpressionParser`:

```
'is' => ['precedence' => 100, 'associativity' => ExpressionParser::OPERATOR_LEFT],
'is not' => ['precedence' => 100, 'associativity' => ExpressionParser::OPERATOR_LEFT],
```

File: src/Test/NodeTestCase.php (hunk 4 above)
Comment (fabpot) [reply src/Test/NodeTestCase.php:56]: It was used when we had more methods reusing the env, but it's not that useful anymore. Maybe we can indeed keep the "cache".

**Pull request merged**